interfaces/builtin/kubernetes_support.go

```diff
@@ -148,6 +148,17 @@ const kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun = `
   /{,usr/}bin/mount ixr,
   mount fstype="tmpfs" tmpfs -> /var/snap/@{SNAP_INSTANCE_NAME}/common/**,
   deny /run/mount/utab rw,
+
+  # For mounting volume subPaths
+  mount /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**} -> /var/snap/@{SNAP_INSTANCE_NAME}/common/{,**},
```

Comment:
do we want to specifically disallow `$SNAP_DATA`? what if someday kubernetes changes to use `$SNAP_DATA` instead of `$SNAP_COMMON` for these volumes?

Reply:
I'm happy to add another commit to for @{SNAP_REVISION}. Let me do that now.

Reply:
probably fine as is then, IIRC for greengrass we made it more open allowing `$SNAP_DATA` or `$SNAP_COMMON`, but that's a bit different as that was a closed source app which could change at any time. k8s is probably more predictable and less likely to change randomly

Reply:
we responded at the same time, either is fine with me, if you already have it ready for snap_revision that's fine, otherwise without is fine too :smile:

Reply:
@anonymouse64 - I added it. Let me know if after reading the commit message you prefer me to revert.

Reply:
After discussion on IRC, I reverted it. The revert's commit message gives the rationale.